hypervisor/arch/x86/sgx.c

```diff
@@ -57,8 +57,9 @@ static int32_t partition_epc(void)
 	uint16_t vm_id = 0U;
 	uint32_t psec_id = 0U, mid = 0U;
 	uint64_t psec_addr = 0UL, psec_size = 0UL;
-	uint64_t vm_request_size = 0UL, free_size = 0UL, alloc_size;
-	struct acrn_vm_config *vm_config;
+	uint64_t free_size = 0UL, alloc_size;
+	struct acrn_vm_config *vm_config = get_vm_config(vm_id);
+	uint64_t vm_request_size = vm_config->epc.size;
 	int32_t ret = 0;
 
 	while ((psec_id < MAX_EPC_SECTIONS) && (vm_id < CONFIG_MAX_VM_NUM)) {
```

Comment:
Could we rewrite the code like this
while (psec_id < ...) {
    if (vm_request_size != 0U) {
        if (free_size == 0UL) {
            ...
        }
        ...
    } else {
        ...
        if (vm_id < ...) {
            ...
        }
    }

}